dataland-backend/src/main/kotlin/org/dataland/datalandbackend/repositories/DataMetaInformationRepository.kt

```diff
@@ -69,4 +70,32 @@ interface DataMetaInformationRepository : JpaRepository<DataMetaInformationEntit
         dataType: String,
         currentlyActive: Boolean,
     ): Long
+
+    /**
+     * Queries the meta information for datasets uploaded by a specific user
+     * @param userId the id of the user for whom to query data meta information
+     * @returns the data meta information uploaded by the specified user
+     */
+    @Query(
+        nativeQuery = true,
+        value = "SELECT " +
+            " datainfo.data_id as dataId," +
+            " datainfo.company_id as companyId," +
+            " company.company_name as companyName," +
+            " datainfo.data_type as dataType," +
+            " datainfo.reporting_period as reportingPeriod," +
+            " datainfo.quality_status as qualityStatus," +
+            " datainfo.currently_active as currentlyActive," +
+            " datainfo.upload_time as uploadTime" +
+            " from (" +
+            " SELECT company_id, data_id, data_type, reporting_period, quality_status, currently_active, upload_time " +
+            " from data_meta_information meta" +
+            " where uploader_user_id = :userId" +
+            " ) datainfo" +
+            " LEFT JOIN stored_companies company" +
+            " ON company.company_id = datainfo.company_id",
+    )
+    fun getUserDataMetaInfos(
```

Comment:
should this be called then getUserUploadsDataMetaInfos, as we not get metadata of the user but other of uploaded data?

Reply:
@fsoffel @BenediktLaubmann  could you also have a look at this?

Reply:
ok was already adressed. closed